esphome/components/growatt_solar/sensor.py

```diff
@@ -152,6 +162,9 @@ async def to_code(config):
     await cg.register_component(var, config)
     await modbus.register_modbus_device(var, config)
 
+    if CONF_PROTOCOL_VERSION in config:
+        cg.add(var.set_protocol_version(config[CONF_PROTOCOL_VERSION]))
+
```

Comment:
Having a default means it's always in the config
```suggestion
    cg.add(var.set_protocol_version(config[CONF_PROTOCOL_VERSION]))

```